scenedetect/detectors/histogram_detector.py

```diff
@@ -108,82 +106,47 @@ def process_frame(self, frame_num: int, frame_img: numpy.ndarray) -> List[int]:
 
         return cut_list
 
-    def _quantize_frame(self, frame_img, bits):
-        """Quantizes the image based on the number of most significant figures to be preserved.
-
-        Arguments:
-            frame_img: The 8-bit rgb image of the frame being analyzed.
-            bits: The number of most significant bits to keep during quantization.
-
-        Returns:
-            [red_img, green_img, blue_img]:
-                The three separated color channels of the frame image that have been quantized.
-        """
-        # First, find the value of the number of most significant bits, padding with zeroes
-        bit_value = int(bin(2**bits - 1).ljust(10, '0'), 2)
-
-        # Separate R, G, and B color channels and cast to int for easier bitwise operations
-        red_img = frame_img[:, :, 0].astype(int)
-        green_img = frame_img[:, :, 1].astype(int)
-        blue_img = frame_img[:, :, 2].astype(int)
-
-        # Quantize the frame images
-        red_img = red_img & bit_value
-        green_img = green_img & bit_value
-        blue_img = blue_img & bit_value
-
-        return [red_img, green_img, blue_img]
-
-    def _shift_bits(self, quantized_imgs, bits):
-        """Takes care of the bit shifting operations to combine the RGB color
-        channels into a single array.
+    def calculate_histogram(self,
+                            frame_img: numpy.ndarray,
+                            normalize: bool = True) -> numpy.ndarray:
+        """Calculates and  normalizes the histogram of the luma (Y) channel of an image converted from BGR to YUV color space.
+            This function extracts the Y channel from the given BGR image, 
+            computes its histogram, and optionally normalizes this histogram to have a sum of one across all bins.
+
+            Args:
+            -----------
+            frame_img : np.ndarray
+                The input image in BGR color space, assumed to have shape (height, width, 3) 
+                where the last dimension represents the BGR channels.
+            normalize : bool, optional (default=True)
+                A boolean flag that determines whether the histogram should be normalized 
+                such that the sum of all histogram bins equals 1.
+
+            Returns:
+            --------
+            np.ndarray
+                A 1D numpy array of length 256, representing the histogram of the luma channel. 
+                Each element in the array represents the count (or frequency) of a particular luma value in the image. 
+                If normalized, these values represent the relative frequency.
+
+            Examples:
+            ---------
+            >>> img = cv2.imread('path_to_image.jpg')
+            >>> hist = calculate_histogram(img, normalize=True)
+            >>> print(hist.shape)
+            (256,)
+       """
+        # Extract Luma channel from the frame image
+        y, _, _ = cv2.split(cv2.cvtColor(frame_img, cv2.COLOR_BGR2YUV))
```

Comment:
Is there a reason to use YUV over something else, say HSV/HSL?  I wonder if it's computationally cheap enough that we do something similar to ContentDetector, and calculate hue/saturation histograms in addition to luma.  That might improve robustness in certain situations.  Thoughts?

Reply:
The idea is that Y channel is not much affected by changes in lighting condition, making it bit more robust than RBG or HSV.
Y channel holds perceptually important information and YCbCr color space is very similar to how eyes perceive light

The color conversion is computationally lightweight since it simple mat diff operation

Reply:
I was unaware that the Y channel isn't as affected during changes in lighting.  Good to know, that may have impact on [another issue in a related project](https://github.com/Breakthrough/DVR-Scan/issues/53).  I think the last thing I would like to check in with then is, is there any value in allowing both color vs. this luma-only method?

Aside from perception of brightness, it's possible that users might require color information as well.  If that's the case we can extend this by allowing the target colorspace to be chosen, and weights placed on each channel.  That can certainly be done as a follow-up however, so feel free to mark this as resolved.